src/omv/imlib/draw.c

```diff
@@ -3144,8 +3144,14 @@ void imlib_draw_image(image_t *dst_img,
             t_src_img.data = src_img->data;
         }
 
+        // Query available on-chip RAM.
         uint32_t size;
         void *data = fb_alloc_all(&size, FB_ALLOC_PREFER_SPEED | FB_ALLOC_CACHE_ALIGN);
+        fb_free();
+
+        // Allocate a buffer to hold chunks of the transposed image while not using all of the on-chip RAM.
+        size = IM_MIN(size, image_size(&t_src_img));
```

Comment:
If the available memory is less than the image size, then this will fail any way right ? So why the query, free, alloc, only to fail ?

Reply:
It can work with less memory. It does the job faster the more memory it has.